### PR TITLE
Integration and sample links

### LINKED ISSUE #9: sample links are removed
Hi,

Sample links are lost while exporting as Sf3. This appears to be done on purpose but I don't know which one.

If it's related to the idea of compressing both left and right samples together, I think it's a bad idea since sf2 format is based on mono samples (played together but 2 distinct mono samples). All sf2 players take into account mono samples to play and if you merge these samples you would force developers to dissociate them first, so that the sf3 format can be read. Two parts of a stereo sample, once imported in a soundfont, may be played completely differently depending on what parameter we put.

### LINKED ISSUE #10: MSVC build
Hi,

I imported the code in Polyphone but when I build it with MSVC:
"char buffer[i];" is not allowed (you can use new char[i] and delete [] instead). This appears at 2 positions in the code.

The line "char buffer[1024*1024];" can be compiled but this crashed the software. Using new char[1048576] and delete [] solved the problem.

### LINKED ISSUE #11: Code improvements
Hi,

Several non-critical things:
* the following dependencies could be avoided if you remove non essential functions: xml module of Qt, libsndfile,
* C++11 could be avoided by rewriting the initialization of a structure,
* warnings could be removed (memset(&p, sizeof(p), 0); / unused variables),
* a namespace is missing,
* readCompressedSample is dead code.

### PR DESCRIPTION
**Pull request by davy7125**
Integration improvement (fix #11):
* preprocessor variable to loose the xml and libsnd dependencies
* namespace SfTools
* smallSf now a class variable
* C++11 not needed anymore (struct initialization)
* dynamic declaration of array now ok with visual C++ (fix #10)
* warning and unused variables removed

Bug fix:
* sample links now kept in the soundfont (fix #9)

### CODE REVIEW

**Comment by davy7125** [comment]
The conversion tool can now convert back from sf3 to sf2
bug fixed: the sm24 chunk doesn't stop the program anymore
improvement: the attenuation is stored in the ogg file so that we can revert it during the extraction

File: sfont.cpp [hunk 1]
```diff
@@ -60,1609 +63,1427 @@ static const bool writeCompressed = true;
...
-      path      = s;
-      engine    = 0;
-      name      = 0;
-      date      = 0;
-      comment   = 0;
-      tools     = 0;
-      creator   = 0;
-      product   = 0;
-      copyright = 0;
-      }
```
Comment (mirabilos) [inline sfont.cpp]: You changed the indentation of everything.

The new indentation is also not the MuseScore indentation style (which is a bit weird, but at least consistently used in MuseScore-related code).

Comment (davy7125) [reply sfont.cpp]: I'm sorry for that I began to edit the files inside my project and I am used to automatically indent files. You could still reopen the files with an IDE or some advanced text editor for reindenting them.

Comment (mirabilos) [reply sfont.cpp]: This makes the diff illegible and therefore impossible to review, though.

Comment (davy7125) [reply sfont.cpp]: I will not blame you if this pull request is closed without being integrated ;-)
The code is working in Polyphone, that's what I wanted

**Comment by mirabilos** [comment]
Davy Triponney dixit:

>I will not blame you if this pull request is closed without being
>integrated ;-)

But I will blame you… I very much *do* want these
fixes and even new features to be merged.

**Comment by mirabilos** [comment]
Hmm, if we have a chance to get this relicenced, maybe there’s a chance you can clean this up (to use the MuseScore coding style to keep the diff legible) so it can be considered for inclusion, and rebase the version in Polyphone on the latest sftools one then?

**Pull request closed without merging**